libbeat/docs/http-endpoint.asciidoc

```diff
@@ -0,0 +1,163 @@
+//////////////////////////////////////////////////////////////////////////
+//// This content is shared by all Elastic Beats. Make sure you keep the
+//// descriptions here generic enough to work for all Beats that include
+//// this file. When using cross references, make sure that the cross
+//// references resolve correctly for any files that include this one.
+//// Use the appropriate variables defined in the index.asciidoc file to
+//// resolve Beat names: beatname_uc and beatname_lc.
+//// Use the following include to pull this content into a doc file:
+//// include::../../libbeat/docs/http-endpoint.asciidoc[]
+//////////////////////////////////////////////////////////////////////////
+
+[[http-endpoint]]
+== HTTP Endpoint
+
+experimental[]
+
+{beatname_uc} can expose internal metrics through an HTTP endpoint. These are useful to
+monitor the internal state of the Beat. For security reasons the endpoint is disabled
+by default, as you may want to avoid exposing this info.
+
+The HTTP endpoint has the following configuration settings:
+
+`http.enabled`:: (Optional) Enable the HTTP endpoint. Default is `false`.
+`http.host`:: (Optional) Bind to this hostname or IP address.
+It is recommended to use only localhost. Default is `localhost`
+`http.port`:: (Optional) Port on which the HTTP endpoint will bind. Default is `5066`.
+
+This is the list of paths you can access. For pretty JSON output append ?pretty to the URL.
+
+
+[float]
+=== Info
+
+`/` provides basic info from the {beatname_uc}. Example:
+
+[source,js]
+----
+curl -XGET 'localhost:5066/?pretty'
+----
+
+["source","js",subs="attributes"]
+----
+{
+  "beat": "metricbeat",
+  "hostname": "example.lan",
+  "name": "example.lan",
+  "uuid": "34f6c6e1-45a8-4b12-9125-11b3e6e89866",
+  "version": "{version}"
+}
+----
+
+[float]
+=== Stats
+
+`/stats` reports internal metrics. Example:
+
+[source,js]
+----
+curl -XGET 'localhost:5066/stats?pretty'
+----
+
+["source","js",subs="attributes"]
+----
+{
```

Comment:
Would be nice if we could auto generate this docs in the future to make sure they stay up-to-date.